core/src/Kokkos_Macros.hpp

```diff
@@ -345,9 +345,9 @@
 #define KOKKOS_ENABLE_RFO_PREFETCH 1
 #endif
 
-#if !defined(KOKKOS_IMPL_FORCEINLINE_FUNCTION)
-#define KOKKOS_IMPL_FORCEINLINE_FUNCTION inline __attribute__((always_inline))
-#define KOKKOS_IMPL_FORCEINLINE __attribute__((always_inline))
+#if !defined(KOKKOS_HOST_FORCEINLINE_FUNCTION)
```

Comment:
Why did you introduce this guy?

Reply:
because `KOKKOS_FORCEINLINE_FUNCTION` includes `__host__ __device__` when CUDA is enabled, and so I can't use it on the methods of the AVX-512 backend because they are host-only

Reply:
Lets make that impl